emission/net/ext_service/transit_matching/match_stops.py

```diff
@@ -35,6 +64,27 @@ def make_request_and_catch(overpass_query):
 
     try:
         all_results = response.json()["elements"]
+        
+        # If we successfully got results and we're not in production, cache them
+        if not in_production and all_results:
+            try:
+                # Convert complex nested structures to strings for CSV storage
+                flat_results = []
+                for result in all_results:
+                    flat_result = result.copy()
+                    # Convert nested objects to strings
+                    for key, value in flat_result.items():
+                        if isinstance(value, (dict, list)):
+                            flat_result[key] = json.dumps(value)
+                    flat_results.append(flat_result)
+                
+                # Create DataFrame and save to CSV
+                df = pd.DataFrame(flat_results)
+                df.to_csv(cache_file, index=False)
+                logging.info(f"Cached API response to {cache_file}")
+            except Exception as e:
+                logging.warning(f"Error caching response: {e}")
```

Comment:
I think you can simplify this. You can stringify the entire API response and store it in one column of the CSV. That way you will not have to treat nested dicts as special cases

If you cannot simplify it significantly then please extract these longer chunks of code to separate functions